R/utils.R

```diff
@@ -465,3 +465,21 @@ infer_sectors_and_technologies <- function(baseline_scenario, shock_scenario, sc
 
   return(list(sectors = shared_sectors, technologies = technologies))
 }
+
+
+infer_scenario_type <- function(baseline_scenario, shock_scenario) {
+  if (grepl("NGFS2021", baseline_scenario) & grepl("NGFS2021", shock_scenario)) {
+    return("is_ngfs")
+  }
+
+  if (!grepl("NGFS2021", baseline_scenario) & !grepl("NGFS2021", shock_scenario)) {
+    return("is_not_ngfs")
+  } else {
+    ((grepl("NGFS2021", baseline_scenario) & !grepl("NGFS2021", shock_scenario)) | (!grepl("NGFS2021", baseline_scenario) & grepl("NGFS2021", shock_scenario)))
+  }
+  rlang::abort(
+    c(
```

Comment:
please refer to other rlang::aborts concerning its structure, e.g. https://github.com/2DegreesInvesting/r2dii.climate.stress.test/blob/ffac2734ee34b460b252fdbfc8b2cba46111177c/R/config-helpers.R#L18